### DISCUSSION TITLE
Issue with GlspSelectionDataService when using with eclipse-glsp/theia-integration: 2.1.0

### DISCUSSION THREAD

**Discussion by amiths89** [eclipse-glsp/glsp#1300] [General] [answered]
I am trying to setup the Theia Properties for the glsp diagram module. I based this upon a sample code in glsp-examples https://github.com/eclipse-glsp/glsp-examples/tree/property-view-example . I had to change the eclipse-glsp/theia-integration version to 2.1.0 to make the GlspSelectionDataService work with the glsp 2.0 code. But now I see an error No matching bindings found for service identifier error message

**Comment by amiths89** [comment]
I was facing an error related to code that is no longer in use in react and to use createRoot instead. For this I updated the GlspSelectionDataService custom code to use the GModelRoot as parameter for which I had to use the eclipse-glsp/theia-integration 2.1.0. if I can get past the react issue while retaining the existing version of packages related to glsp that would seem like a better fix

**Comment by tortmayr** [comment]
@ndoschek Could you have a look at this. I'm not familiar with the properties-view-example and it seems like there are some adaptions necessary to make this work with newer GLSP versions

**Reply by ndoschek** [reply]
Sure I'll look into it by the end of the week!

**Comment by ndoschek** [comment, marked as answer]
Hi @amiths89,

I just updated the property-view example (see https://github.com/eclipse-glsp/glsp-examples/tree/property-view-example) to GLSP version 2.1.0 and it works fine on my side.
Could you please have a look if it resolves the issues for you as well?
If not, could you give me more details about the errors you get?

HTH and best regards,
Nina

**Reply by amiths89** [reply]
Thanks @ndoschek for updating the code to 2.1. I was able to run the example. Although I'm still not able to make it work for my code. I currently have 2 diagrams running. With the property code code implemented am not able to open the diagram extension files. I have no issues building and starting theia but once its started the diagrams does not seem to be working

**Reply by amiths89** [reply]
@ndoschek Sorry i think your solution works perfectly. The problem seems to be something else. I have feature for snapper which seems to be updated in the 2.1.0. So when i took it out the property seems to be fine. It looks like snapper does not work the same way as it used to in 2.0.0.

Thank you again @ndoschek. Also my goal is to be able to edit here, is it possible to edit the node data using theia property view?  or will i need to look at UI extensions and json forms?

**Reply by ndoschek** [reply]
Hi @amiths89,

Regarding editing property values, you can use the `onChange` hook from the property view widget.
To do so you can extend the widget provider (`tasklist-theia/src/browser/property-view/custom-widget-provider.ts`) as mentioned below. It depends of course how you want to update your model.

```typescript
import { JsonFormsPropertyViewWidgetProvider } from '@eclipse-emfcloud/jsonforms-property-view';
import { GlspSelection } from '@eclipse-glsp/theia-integration';
import { injectable, postConstruct } from 'inversify';
import { debounce } from 'lodash';

@injectable()
export class GlspPropertyViewWidgetProvider extends JsonFormsPropertyViewWidgetProvider {
    @postConstruct()
    override init(): void {
        super.init();
        this.registerWidgetChangeHandler();
    }

    protected registerWidgetChangeHandler(): void {
        this.jsonFormsWidget.onChange(
            debounce((jsonFormsData: object) => {
                this.handleChanges(jsonFormsData);
            }, 250)
        );
    }

    override canHandle(selection: any): number {
        return GlspSelection.is(selection) ? 1 : 0;
    }

    protected handleChanges(jsonFormsData: object | undefined): void {
        // handle the changed form data
        // e.g. change model via GLSP actions or your custom model management
    }
}
```

I also added a separate commit with this snippet to the initial example branch.

Regarding the snapper issue, I'll look into it if I can spot anything major that has changed.

HTH and best regards,

Nina

**Reply by amiths89** [reply]
Thanks a lot @ndoschek. I will try this and the commit. Oh and regarding the snapper I got around the issue I needed to make the changes to adjust with the PositionSnapper.
